packages/x-scheduler/src/joy/event-calendar/store.ts

```diff
@@ -0,0 +1,53 @@
+import { getAdapter } from '../../primitives/utils/adapter/getAdapter';
+import { createSelector, createSelectorMemoized, Store } from '../../base-ui-copy/utils/store';
+import { SchedulerValidDate } from '../../primitives/models';
+import { CalendarEvent } from '../models/events';
+import { CalendarResource, CalendarResourceId } from '../models/resource';
+import { ViewType } from '../models/views';
+
+const adapter = getAdapter();
+
+export type State = {
+  visibleDate: SchedulerValidDate;
+  currentView: ViewType;
+  views: ViewType[];
+  events: CalendarEvent[];
+  resources: CalendarResource[];
+};
+
+export type EventCalendarStore = Store<State>;
+
+export const selectors = {
+  visibleDate: createSelector((state: State) => state.visibleDate),
+  currentView: createSelector((state: State) => state.currentView),
+  views: createSelector((state: State) => state.views),
+  resources: createSelector((state: State) => state.resources),
+  resourcesByIdMap: createSelectorMemoized(
+    (state: State) => state.resources,
+    (resources) => {
+      const map = new Map<CalendarResourceId | undefined, CalendarResource>();
+      for (const resource of resources) {
+        map.set(resource.id, resource);
+      }
+      return map;
+    },
+  ),
+  getEventsStartingInDay: createSelectorMemoized(
```

Comment:
We can probably have a stronger data structure at some point to avoid computing the map for all the days even the one not used in the views.
But I think this simple approach will scale well until we start discussing lazy loading and has the major benefit of making the access of the data super simple.